tableaudocumentapi/datasource.py

```diff
@@ -258,3 +261,84 @@ def _get_metadata_objects(self):
     def _get_column_objects(self):
         return [_column_object_from_column_xml(self._datasourceTree, xml)
                 for xml in self._datasourceTree.findall('.//column')]
+
+    def add_field(self, name, datatype, role, type, caption):
+        """ Adds a base field object with the given values.
+
+        Args:
+            name: Name of the new Field. String.
+            datatype:  Datatype of the new field. String.
+            role:  Role of the new field. String.
+            type:  Type of the new field. String.
+            caption:  Caption of the new field. String.
+
+        Returns:
+            The new field that was created. Field.
+        """
+        # TODO: A better approach would be to create an empty column and then
+        # use the input validation from its "Field"-object-representation to set values.
+        # However, creating an empty column causes errors :(
```

Comment:
The Field objects __init__() expects some sort of XML as input. Currently, it's not possible to initialize an "empty" Field and fill it with values later on.